airbyte-cdk/bulk/toolkits/extract-cdc/src/main/kotlin/io/airbyte/cdk/read/cdc/Debezium.kt

```diff
@@ -85,18 +85,19 @@ value class DebeziumRecordValue(val wrapped: JsonNode) {
     }
 }
 
-/** Debezium Engine input. */
-data class DebeziumInput(
-    val properties: Map<String, String>,
-    val state: DebeziumState,
-    val isSynthetic: Boolean,
-)
-
-/** Debezium Engine output, other than records of course. */
-data class DebeziumState(
+/** Return type for [CdcPartitionsCreatorDebeziumOperations.deserializeState]. */
+sealed interface DebeziumWarmStartState
+
+data class ValidDebeziumWarmStartState(
     val offset: DebeziumOffset,
-    val schemaHistory: DebeziumSchemaHistory?,
-)
+    val schemaHistory: DebeziumSchemaHistory?
+) : DebeziumWarmStartState
+
+sealed interface InvalidDebeziumWarmStartState : DebeziumWarmStartState
+
+data class AbortDebeziumWarmStartState(val reason: String) : InvalidDebeziumWarmStartState
+
+data class ResetDebeziumWarmStartState(val reason: String) : InvalidDebeziumWarmStartState
```

Comment:
This abort vs reset failure handling is going to be present in every connector with CDC if it isn't the case already, so it makes sense to recognize it in the CDK. Previously `OffsetInvalidNeedsResyncIllegalStateException` was used as a stopgap but using exceptions for control flow can be confusing.